R/tar_download.R

```diff
@@ -0,0 +1,104 @@
+#' @title Download a file from a remote source, checking for changes
+#'   first
+#' @description Create a target that downnloads a file if it has changed since
+#'   the last attempt, based responds to a change
+#'   in an arbitrary value. If the value changes, the target reruns.
+#' @details `tar_download()` creates a pair of targets, one upstream
+#'   and one downstream, usingn `tar_change()`. always runs and returns
+#'   header information (eTag and modified time, if available), to check
+#'   if the remote value has changed. This header gets referenced in the
+#'   downstream target, which causes the downstream target to download the
+#'   if the header information changes. The downstream target has "file" format
+#'   and stores the location of te downloaded file The behavior is cancelled if
+#'   `cue` is `tar_cue(depend = FALSE)` or `tar_cue(mode = "never")`.
+#' @export
+#' @return A list of two targets, one upstream and one downstream.
+#'   The upstream one checks url headers, and the downstream one
+#'   downloads the file to it. See the examples for details.
+#' @param url The URL to fetch the file from
+#' @param destfile Whether to invoke tidy evaluation
+#' @param destdir the directory to download the file to 
+#' @param stop_on_no_internet Whether to fail in the absence of an internet connection.
+#' Default is `FALSE` and only produces a warning
+#' @param handle a handle created by `curl::new_handle()` to use when fetching the file
+#'   or checking headers. Useful for authentication.
+#' @param ... other parameters passed to `tar_change()`
+#' @examples
+#' \dontrun{
+#' 
+# targets::tar_dir({
+# targets::tar_script({
+#   library(tarchetypes)
+#   tar_pipeline(
+#     tar_download(gitfile, url = "https://github.com/wlandau/tarchetypes/archive/master.zip")
+#   )
+# })
+# targets::tar_make(callr_function = NULL)
+# })
+#' }
+tar_download <- function(
+  name,
+  url,
+  destfile = basename(url),
+  destdir = ".",
+  handle = NULL,
+  stop_on_no_internet = FALSE,
+  ...
+) {
+  if(!requireNamespace("curl", quietly = TRUE)) {
```

Comment:
Would you use `tarchetypes::assert_package("curl", "tar_download() requires the the package 'curl' to be installed")`?